### PR TITLE
#2466 ElementMatcher no longer clears the slicing element of an extension element when the match action is slice

### PR DESCRIPTION
**Pull request by Rob5045**
## Related issues
Fixes issue #2466 

## Testing
Added unit test TestBaseAnnotations_StructureDefinition_ExtensionHeaderSlicingElement and updated TestBaseAnnotations_ExplicitCoreTypeProfile.

### CODE REVIEW

File: src/Hl7.Fhir.Specification.Shared.Tests/Snapshot/SnapshotGeneratorTest.cs [hunk 1]
```diff
@@ -2161,6 +2159,92 @@ public async T.Task TestBaseAnnotations_InlineExtension()
...
+                FhirVersion = FHIRVersion.N4_0_1,
+                Derivation = StructureDefinition.TypeDerivationRule.Constraint,
+                Kind = StructureDefinition.StructureDefinitionKind.Resource,
+                Differential = new StructureDefinition.DifferentialComponent()
+                {
+                    Element = new List<ElementDefinition>()
+                    {
+                        new ElementDefinition(path)
+                        {
+                            Min = 0
```
Comment (marcovisserFurore) [inline src/Hl7.Fhir.Specification.Shared.Tests/Snapshot/SnapshotGeneratorTest.cs:2186]: This constraint is probably not needed.

**Review by marcovisserFurore — CHANGES_REQUESTED** [review]
I think we should have this solution for STU3 as well.

File: Hl7.Fhir.sln
Comment (Rob5045) [inline Hl7.Fhir.sln:1]: Visual Studio changed the solution file for me and I could not undo it!

Comment (marcovisserFurore) [reply Hl7.Fhir.sln:1]: Can you undo this in git?

Comment (Rob5045) [reply Hl7.Fhir.sln:1]: I will try

Comment (Rob5045) [reply Hl7.Fhir.sln:1]: Done

**Review by marcovisserFurore — APPROVED** [review]

**Pull request merged**